doc/static-route/SONiC_static_route_bfd_hld.md

```diff
@@ -0,0 +1,300 @@
+
+To avoid StaticRouteTimer deleting StaticRouteBfd created static route entry in appl_db, a new field "expiry" is introduced in appl_db STATIC_ROUTE_TABLE schema. StaticRouteBfd sets "expiry"="false" when it writes a static route to appl_db STATIC_ROUTE_TABLE. When StaticRouteTimer see "expiry"="false", it ignores this static route entry. <br>
+
+
+<img src="static_rt_bfd_overview.png" width="500">
+<br />
+
+## BFD session local IP address
+StaticRouteBfd create BFD session for each nexthop inside a BFD enabled static route. The BFD peer IP is the nexthop, the BFD local address (source IP address of the BFD packet) is the lookup result using the following method:<br />
+
+1. find the interface name for the nexthop
+2. lookup the config_db, find the IP address of that interface
+3. choose the IP address which has same type of nexthop (IPv4 or IPv6)
+4. If no IP address found, use local loopback IP address, and log a warning message for that. <br /><br />
+
+In the example (assume the "bfd" field is "true"), for the nexthop "20.0.10.3", the corresponding  interface name is PortChannel10.
+From the interface configuration, we can found two IP addresses, an IPv4 address 20.0.10.1 and an IPv6 address 2603:10E2:400:10::1. we choose 20.0.10.1 because the nexthop IP address is IPv4 address. <br /><br />
+
+<img src="static_rt_bfd_cfg.png" width="800">
+<img src="static_rt_bfd_interface.png" width="500">
+<br />
+
+## DB changes
+ 
+Two optional fields are introduced to STATIC_ROUTE_TABLE: <br>
+1. "bfd" -- Default value is false when this field is not configured
+2. "expiry" -- Default value is true when this field is not configured
+
+StaticRouteMgr(config_db) and StaticRouteBfd check "bfd" field in config_db STATIC_ROUTE_TABLE.<br>
+StaticRouteBfd sets "expiry"="false" in appl_db STATIC_ROUTE_TABLE. 
+StaticRouteTimer checks "expiry" field to skip the static route entry timeout checking.<br>
+
+[*Reference: STATIC_ROUTE_TABLE schema:* 
+ [STATIC_ROUTE table in CONFIG_DB](https://github.com/Azure/SONiC/blob/master/doc/static-route/SONiC_static_route_hdl.md#3211-static_route).]
+
+
+## Internal tables in StaticRouteBfd
+Four tables (i.e., dictionary, map, etc) are needed to use BFD session to monitor nexthop and update static route.<br>
+
+1. TABLE_CONFIG: config_db STATIC_ROUTE_TABLE cache (for the route with "bfd"="true" only)
+2. TABLE_NEXTHOP: different prefixes may have same nexthop. This table is used to track which prefix is using the nexthop.
+3. TABLE_BFD: bfd session created by StaticRouteBfd. The contents are part of appl_db BFD_SESSION_TABLE (for the session its peer IP is in nexthop table). *Assumption: BFD session is not shared with other applications*
+4. TABLE_SR: the static routes written to appl_db STATIC_ROUTE_TABLE by BfdRouteManager (with "expiry"="false"). It's nexthop list might be different from the configuration depends on BFD session state.<br>
```

Comment:
Suggest to STATIC_ROUTE instead of SR as it conflicts with segment routing.

Reply:
Changed the table name to TABLE_SRT (avoiding SR, but want to keep the name short).